src/background/in-dapp-notifications.ts

```diff
@@ -0,0 +1,108 @@
+import { getActiveTabOrigin } from 'src/ui/shared/requests/getActiveTabOrigin';
+import { networksStore } from 'src/modules/networks/networks-store.background';
+import type { ChainId } from 'src/modules/ethereum/transactions/ChainId';
+import type { Chain } from 'src/modules/networks/Chain';
+import { emitter } from './events';
+
+interface Notification {
+  title: string;
+  message: string;
+  icon: string;
+  compact: boolean;
+}
+
+declare global {
+  interface Window {
+    createNotification(notification: Notification): void;
+  }
+}
+
+// We can't directly pass the window.createNotification to the chrome.scripting.executeScript,
+// so we have to create a separate wrapper function. More info: https://developer.chrome.com/docs/extensions/reference/api/scripting#type-ScriptInjection
+function showNotification(notification: Notification) {
+  window.createNotification(notification);
```

Comment:
Not sure do we need to check here `if(window.createNotification)` or something
Just to avoid error in console. However, this case will be broken anyway

Reply:
not relevant anymore, I updated the implementation to get rid of `window.createNotification`